geoutils/georaster.py

```diff
@@ -822,10 +913,179 @@ def format_value(value):
             raise ValueError(
                 'Value provided for band was not int or None.')
 
-        if return_window == True:
+        if return_window:
             return (value, win)
         else:
             return value
-    
+
+    def xxyy(self, offset='corner', grid=True):
+        """
+        Get x,y coordinates of all pixels in the raster.
+
+        :param offset: coordinate type. If 'corner', returns corner coordinates of pixels.
+            If 'center', returns center coordinates. Default is corner.
+        :param grid: Return gridded coordinates. Default is True.
+        :type offset: str
+        :type grid: bool
+        :returns x,y: numpy arrays corresponding to the x,y coordinates of each pixel.
+        """
+        assert offset in ['corner', 'center'], "ctype is not one of 'corner', 'center': {}".format(offset)
+
+        xmin, ymin, xmax, ymax = self.ds.bounds
```

Comment:
Here and everywhere, you should use the attributes of self, rather than self.ds, for example: self.bounds, self.width, self.height etc in case self.ds is closed.
Also dx and dy are already available through self.res

Reply:
Also better to stay consistent with names, use height and width instead of 'npix_x' and 'npix_y'.

Reply:
There's also an error, should dx not be list(self.ds.transform)[1]?

Reply:
Agreed, this can be shortened.
No error I think, the transform provided through rio looks like this: [30.0, 0.0, 478000.0, 0.0, -30.0, 3108140.0, 0.0, 0.0, 1.0]

Reply:
Ok, so rio's transform is different from GDAL's, good to know!

Reply:
Yeah I fell into the same trap and wrote [1] and [5] at first. I'm so used to GDAL, switching is a challenge... é_è